core/server/data/migrations/versions/5.0/2022-05-11-13-12-settings-rename.js

```diff
@@ -0,0 +1,43 @@
+const logging = require('@tryghost/logging');
+const {createTransactionalMigration} = require('../../utils');
+
+const renameMappings = [{
+    from: 'lang',
+    to: 'locale'
+}, {
+    from: 'session_secret',
+    to: 'admin_session_secret'
+}];
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const renameMapping of renameMappings) {
+            const oldSetting = await knex('settings')
+                .where('key', renameMapping.from)
+                .select('value')
+                .first();
+
+            if (!oldSetting) {
+                logging.warn(`Could not find setting ${renameMapping.from}, not updating ${renameMapping.to} value`);
+                continue;
+            }
+
+            const updatedValue = renameMapping.getToValue ? renameMapping.getToValue(oldSetting.value) : oldSetting.value;
+
+            logging.info(`Updating ${renameMapping.to} with value from ${renameMapping.from}`);
+            await knex('settings')
+                .where('key', renameMapping.to)
+                .update('value', updatedValue);
```

Comment:
I think this should be an insert, because afaict we'll never have a row with `key: renameMapping.to` at this point so we'd just be deleting settings 😬